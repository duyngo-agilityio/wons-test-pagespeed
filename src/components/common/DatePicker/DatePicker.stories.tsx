// Libs
import type { Meta, StoryObj } from '@storybook/react';
import { parseAbsoluteToLocal } from '@internationalized/date';

// Components
import DatePicker from '.';

const meta: Meta<typeof DatePicker> = {
  title: 'Components/Common/DatePicker',
  component: DatePicker,
  args: {},
  argTypes: {
    label: {
      description: 'The content to display as the label.',
      type: 'string',
    },
    defaultValue: {
      label: 'The default value of the date-picker ',
      type: 'string',
    },
    isInvalid: {
      description: 'Whether the date-picker is invalid.',
      type: 'boolean',
    },
    errorMessage: {
      description: 'An error message for the date input.',
      type: 'string',
    },
    granularity: {
      description:
        'Determines the smallest unit that is displayed in the date picker. Typically "day" for dates.',
      type: 'string',
      options: ['day', 'hour', 'minute', 'second'],
      control: { type: 'select' },
    },
    className: {
      description: 'Change style as you like',
      type: 'string',
    },
  },
};

export default meta;
type Story = StoryObj<typeof DatePicker>;

export const Default: Story = {
  args: {
    className: 'w-[300px]',
  },
};

export const HasValue: Story = {
  args: {
    label: 'Date',
    defaultValue: parseAbsoluteToLocal('2024-09-07T18:45:22Z'),
    granularity: 'day',
    className: 'w-[300px]',
  },
};

export const HasError: Story = {
  args: {
    label: 'Date',
    defaultValue: parseAbsoluteToLocal('2024-09-07T18:45:22Z'),
    granularity: 'day',
    isInvalid: true,
    errorMessage: 'Value must be 9/3/2024 or later.',
    className: 'w-[300px]',
  },
};
