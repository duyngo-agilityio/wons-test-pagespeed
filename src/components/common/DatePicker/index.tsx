'use client';

// Libs
import React from 'react';
import {
  DatePicker as DatePickerNextUI,
  DatePickerProps,
} from '@nextui-org/react';

// Components
import { FaCalendarAlt } from '@/components/common';

const DatePicker = ({ ...props }: DatePickerProps) => {
  return (
    <DatePickerNextUI
      labelPlacement="outside"
      endContent={<FaCalendarAlt className="w-[18px] h-5 text-blue-500" />}
      dateInputClassNames={{
        inputWrapper: [
          'px-5 pt-[17px] pb-[14px] h-12.5',
          'rounded-10',
          'bg-gray-50 dark:bg-gray-600',
        ],
        input: ['text-md font-dm-sans', 'color-blue-800 dark:color-white'],
        label: ['text-xl font-medium font-dm-sans'],
      }}
      {...props}
    />
  );
};

export default DatePicker;
