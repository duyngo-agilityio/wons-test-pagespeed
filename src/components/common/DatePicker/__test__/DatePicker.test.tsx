import { DatePickerProps } from '@nextui-org/react';

// Components
import DatePicker from '..';

describe('DatePicker component', () => {
  const renderComponent = (props?: DatePickerProps) =>
    testLibJestUtils.render(<DatePicker {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match with snapshot', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
