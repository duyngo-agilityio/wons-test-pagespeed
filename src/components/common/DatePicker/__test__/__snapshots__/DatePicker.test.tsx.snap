// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component should match with snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="group flex flex-col data-[has-helper=true]:pb-[calc(theme(fontSize.tiny)_+8px)] gap-y-1.5 group w-full"
      data-has-end-content="true"
      data-slot="base"
    >
      <div
        aria-describedby="react-aria-description-1"
        aria-labelledby="react-aria-:r3:"
        class="relative gap-3 inline-flex flex-row items-center cursor-text tap-highlight-transparent shadow-sm hover:bg-default-200 focus-within:hover:bg-default-100 group-data-[invalid=true]:bg-danger-50 group-data-[invalid=true]:hover:bg-danger-100 group-data-[invalid=true]:focus-within:hover:bg-danger-50 min-h-10 rounded-medium w-full transition-background motion-reduce:transition-none !duration-150 px-5 pt-[17px] pb-[14px] h-12.5 rounded-10 bg-gray-50 dark:bg-gray-600"
        data-slot="input-wrapper"
        id="react-aria-:r3:"
        role="group"
      >
        <div
          class="flex items-center text-default-400 w-full gap-x-2 h-6 group-data-[invalid=true]:text-danger"
          data-slot="inner-wrapper"
        >
          <div
            class="flex h-full gap-x-0.5 w-full font-normal text-md font-dm-sans color-blue-800 dark:color-white"
            data-slot="input-field"
            id="react-aria-:r2:"
            role="presentation"
          >
            <div
              aria-describedby="react-aria-description-1"
              aria-label="month"
              aria-valuemax="12"
              aria-valuemin="1"
              aria-valuenow="9"
              aria-valuetext="09 â€“ September"
              autocorrect="off"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              contenteditable="true"
              data-editable="true"
              data-slot="segment"
              data-type="month"
              enterkeyhint="next"
              id="react-aria-:rd:"
              inputmode="numeric"
              role="spinbutton"
              spellcheck="false"
              style="caret-color: transparent;"
              tabindex="0"
            >
              09
            </div>
            <div
              aria-hidden="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="literal"
            >
              /
            </div>
            <div
              aria-label="day"
              aria-valuemax="30"
              aria-valuemin="1"
              aria-valuenow="8"
              aria-valuetext="08"
              autocorrect="off"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              contenteditable="true"
              data-editable="true"
              data-slot="segment"
              data-type="day"
              enterkeyhint="next"
              id="react-aria-:rh:"
              inputmode="numeric"
              role="spinbutton"
              spellcheck="false"
              style="caret-color: transparent;"
              tabindex="0"
            >
              08
            </div>
            <div
              aria-hidden="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="literal"
            >
              /
            </div>
            <div
              aria-label="year"
              aria-valuemax="9999"
              aria-valuemin="1"
              aria-valuenow="2024"
              aria-valuetext="2024"
              autocorrect="off"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              contenteditable="true"
              data-editable="true"
              data-slot="segment"
              data-type="year"
              enterkeyhint="next"
              id="react-aria-:rl:"
              inputmode="numeric"
              role="spinbutton"
              spellcheck="false"
              style="caret-color: transparent;"
              tabindex="0"
            >
              2024
            </div>
            <div
              aria-hidden="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="literal"
            >
              , 
            </div>
            <div
              aria-label="hour"
              aria-valuemax="11"
              aria-valuemin="0"
              aria-valuenow="1"
              aria-valuetext="1 AM"
              autocorrect="off"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              contenteditable="true"
              data-editable="true"
              data-slot="segment"
              data-type="hour"
              enterkeyhint="next"
              id="react-aria-:rp:"
              inputmode="numeric"
              role="spinbutton"
              spellcheck="false"
              style="caret-color: transparent;"
              tabindex="0"
            >
              01
            </div>
            <div
              aria-hidden="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="literal"
            >
              :
            </div>
            <div
              aria-label="minute"
              aria-valuemax="59"
              aria-valuemin="0"
              aria-valuenow="45"
              aria-valuetext="45"
              autocorrect="off"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              contenteditable="true"
              data-editable="true"
              data-slot="segment"
              data-type="minute"
              enterkeyhint="next"
              id="react-aria-:rt:"
              inputmode="numeric"
              role="spinbutton"
              spellcheck="false"
              style="caret-color: transparent;"
              tabindex="0"
            >
              45
            </div>
            <div
              aria-hidden="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="literal"
            >
              â€¯
            </div>
            <div
              aria-label="AM/PM"
              aria-valuemax="12"
              aria-valuemin="0"
              aria-valuenow="0"
              aria-valuetext="AM"
              autocorrect="off"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              contenteditable="true"
              data-editable="true"
              data-slot="segment"
              data-type="dayPeriod"
              enterkeyhint="next"
              id="react-aria-:r11:"
              role="spinbutton"
              spellcheck="false"
              style="caret-color: transparent;"
              tabindex="0"
            >
              AM
            </div>
            <div
              aria-hidden="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="literal"
            >
               
            </div>
            <div
              aria-label="time zone"
              aria-readonly="true"
              class="group first:-ml-0.5 [&:not(:first-child)]:-ml-1 px-0.5 my-auto box-content tabular-nums text-start inline-block outline-none focus:shadow-sm rounded-md text-foreground-500 data-[editable=true]:text-foreground data-[editable=true]:data-[placeholder=true]:text-foreground-500 data-[invalid=true]:text-danger-300 data-[invalid=true]:data-[editable=true]:text-danger data-[invalid=true]:focus:bg-danger-400/50 dark:data-[invalid=true]:focus:bg-danger-400/20 data-[invalid=true]:data-[editable=true]:focus:text-danger focus:bg-default-400/50 data-[editable=true]:focus:text-default-foreground transition-colors motion-reduce:transition-none"
              data-slot="segment"
              data-type="timeZoneName"
              id="react-aria-:r15:"
              role="textbox"
              style="caret-color: transparent;"
              tabindex="0"
            >
              GMT+7
            </div>
            <input
              type="hidden"
              value="2024-09-08T01:45:22+07:00[Asia/Bangkok]"
            />
          </div>
          <button
            aria-describedby="react-aria-description-1"
            aria-expanded="false"
            aria-haspopup="dialog"
            aria-label="Calendar"
            aria-labelledby="react-aria-:r0: react-aria-:r3:"
            class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny gap-2 rounded-full px-0 !gap-0 transition-transform-colors-opacity motion-reduce:transition-none bg-transparent data-[hover=true]:bg-default/40 min-w-8 w-8 h-8 -mr-2 text-inherit"
            data-slot="selector-button"
            id="react-aria-:r0:"
            type="button"
          >
            <svg
              class="w-[18px] h-5 text-blue-500"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
